### ISSUE TITLE
Build never ends

### ISSUE THREAD

**Issue by jeusdi** [openshift/origin#20391]
I'm facing a problem.

```
$ oc v3.9.0+191fece
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://master.si-origin-cluster.t-systems.es:8443
openshift v3.9.0+ba7faec-1
kubernetes v1.9.1+a0ce1bc657
```

I'm performing:

    mvn -Dfabric8.namespace=t-devhub-dev \
      -Dfabric8.openshift.trimImageInContainerSpec=true \
      -Dmaven.test.skip=true \
      clean package fabric8:resource fabric8:build

This process is hanging on waiting for `fabric8:build` ends:

    [INFO] --- fabric8-maven-plugin:3.5.38:build (default-cli) @ tdev-forum-service ---
    [INFO] F8: Using OpenShift build with strategy Docker
    [INFO] Copying files to /home/jcabre/projects/tdevhub/application-src/t-devhub/tdev-forum-service/target/docker/forum/0.0.1-SNAPSHOT/build/maven
    [INFO] Building tar: /home/jcabre/projects/tdevhub/application-src/t-devhub/tdev-forum-service/target/docker/forum/0.0.1-SNAPSHOT/tmp/docker-build.tar
    [INFO] F8: [forum:0.0.1-SNAPSHOT]: Created docker source tar /home/jcabre/projects/tdevhub/application-src/t-devhub/tdev-forum-service/target/docker/forum/0.0.1-SNAPSHOT/tmp/docker-build.tar
    [INFO] F8: Updating BuildServiceConfig forum-s2i for Docker strategy
    [INFO] F8: Adding to ImageStream forum
    [INFO] F8: Starting Build forum-s2i

`docker-build.tar` is really straigforward:

    ├── Dockerfile
    ├── maven
    │   └── tdev-forum-service-0.0.1-SNAPSHOT.jar
    └── .s2i
        └── environment

where `Dockerfile` is:

    FROM docker.io/fabric8/java-jboss-openjdk8-jdk:1.2
    ENV JAVA_APP_JAR=tdev-forum-service-0.0.1-SNAPSHOT.jar
    COPY maven /deployments/

and `.s2i/environment`:

    JAVA_APP_JAR=tdev-forum-service-0.0.1-SNAPSHOT.jar

Also, I've tried using `oc` client:

    cat docker-build.tar | oc start-build --from-dir=- --loglevel=10 forum-s2i --follow

It's also handing on:

    I0723 12:17:06.770656    3403 loader.go:357] Config loaded from file /home/jcabre/.kube/config
    Uploading archive file from STDIN as binary input for the build ...
    round_trippers.go:417] curl -k -v -XPOST  -H "Accept: application/json, */*" -H "User-Agent: oc/v1.9.1+a0ce1bc657 (linux/amd64) kubernetes/a0ce1bc" -H "Authorization: Bearer cQFjgh_mjoAkbjYRhyz0sYCr8SaeauIpi4cjYL2UzDY" https://<cluster-url>/v1/namespaces/t-devhub-dev/buildconfigs/forum-s2i/instantiatebinary?name=forum-s2i&namespace=t-devhub-dev

It's currently happing with all my builds (this output is for `tdev-wsec-service`):

`oc describe build tdev-wsec-service-s2i` output:

```
Name:		tdev-wsec-service-s2i-1
Namespace:	t-devhub-dev
Created:	28 minutes ago
Labels:		app=tdev-wsec-service
		buildconfig=tdev-wsec-service-s2i
		group=org.springframework.boot
		openshift.io/build-config.name=tdev-wsec-service-s2i
		openshift.io/build.start-policy=Serial
		provider=fabric8
		version=2.0.3.RELEASE
Annotations:	openshift.io/build-config.name=tdev-wsec-service-s2i
		openshift.io/build.number=1
		openshift.io/build.pod-name=tdev-wsec-service-s2i-1-build

Status:		Running
Started:	Mon, 23 Jul 2018 12:06:31 CEST
Duration:	running for 28m25s

Build Config:	tdev-wsec-service-s2i
Build Pod:	tdev-wsec-service-s2i-1-build

Strategy:	Docker
Output to:	ImageStreamTag tdev-wsec-service:2.0.3.RELEASE
Binary:		provided on build
Push Secret:	builder-dockercfg-t8ktw

Build trigger cause:	<unknown>

Events:
  Type		Reason			Age	From						Message
  ----		------			----	----						-------
  Normal	SuccessfulMountVolume	28m	kubelet, node4.si-origin-cluster.t-systems.es	MountVolume.SetUp succeeded for volume "builder-dockercfg-t8ktw-push"
  Normal	Scheduled		28m	default-scheduler				Successfully assigned tdev-wsec-service-s2i-1-build to node4.si-origin-cluster.t-systems.es
  Normal	SuccessfulMountVolume	28m	kubelet, node4.si-origin-cluster.t-systems.es	MountVolume.SetUp succeeded for volume "docker-socket"
  Normal	SuccessfulMountVolume	28m	kubelet, node4.si-origin-cluster.t-systems.es	MountVolume.SetUp succeeded for volume "crio-socket"
  Normal	SuccessfulMountVolume	28m	kubelet, node4.si-origin-cluster.t-systems.es	MountVolume.SetUp succeeded for volume "buildworkdir"
  Normal	SuccessfulMountVolume	28m	kubelet, node4.si-origin-cluster.t-systems.es	MountVolume.SetUp succeeded for volume "builder-token-fz1d5"
  Normal	Pulled			28m	kubelet, node4.si-origin-cluster.t-systems.es	Container image "openshift/origin-docker-builder:v3.9.0" already present on machine
  Normal	Created			28m	kubelet, node4.si-origin-cluster.t-systems.es	Created container
  Normal	Started			28m	kubelet, node4.si-origin-cluster.t-systems.es	Started container
  Normal	BuildStarted		28m	build-controller				Build t-devhub-dev/tdev-wsec-service-s2i-1 is now running
```

Logs:

`oc logs tdev-wsec-service-s2i-1-build`

`Error from server (BadRequest): container "docker-build" in pod "tdev-wsec-service-s2i-1-build" is waiting to start: PodInitializing`

Here it goes:

```
$ oc describe pod tdev-wsec-service-s2i-1-build
Name:           tdev-wsec-service-s2i-1-build
Namespace:      t-devhub-dev
Node:           node4.si-origin-cluster.t-systems.es/10.49.160.23
Start Time:     Mon, 23 Jul 2018 12:06:31 +0200
Labels:         openshift.io/build.name=tdev-wsec-service-s2i-1
Annotations:    openshift.io/build.name=tdev-wsec-service-s2i-1
                openshift.io/scc=privileged
Status:         Pending
IP:             10.130.3.0
Controlled By:  Build/tdev-wsec-service-s2i-1
Init Containers:
  git-clone:
    Container ID:  docker://31ad6ece960d071eb7b19323ffdbd029062e6805dc616309abbafd0af2815b36
    Image:         openshift/origin-docker-builder:v3.9.0
    Image ID:      docker-pullable://docker.io/openshift/origin-docker-builder@sha256:6e57da79f6fa2bb99319722d02136ab2a4b5eba41699b8a17209180c0101596a
    Port:          <none>
    Command:
      openshift-git-clone
    Args:
      --loglevel=0
    State:          Running
      Started:      Mon, 23 Jul 2018 12:06:32 +0200
    Ready:          False
    Restart Count:  0
    Environment:
      BUILD:  {"kind":"Build","apiVersion":"v1","metadata":{"name":"tdev-wsec-service-s2i-1","namespace":"t-devhub-dev","selfLink":"/apis/build.openshift.io/v1/namespaces/t-devhub-dev/builds/tdev-wsec-service-s2i-1","uid":"1291eeec-8e60-11e8-b1f7-0050569fe304","resourceVersion":"50777263","creationTimestamp":"2018-07-23T10:06:31Z","labels":{"app":"tdev-wsec-service","buildconfig":"tdev-wsec-service-s2i","group":"org.springframework.boot","openshift.io/build-config.name":"tdev-wsec-service-s2i","openshift.io/build.start-policy":"Serial","provider":"fabric8","version":"2.0.3.RELEASE"},"annotations":{"openshift.io/build-config.name":"tdev-wsec-service-s2i","openshift.io/build.number":"1"},"ownerReferences":[{"apiVersion":"build.openshift.io/v1","kind":"BuildConfig","name":"tdev-wsec-service-s2i","uid":"11dea374-8e60-11e8-b1f7-0050569fe304","controller":true}]},"spec":{"serviceAccount":"builder","source":{"type":"Binary","binary":{}},"strategy":{"type":"Docker","dockerStrategy":{"env":[{"name":"HTTP_PROXY","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"HTTPS_PROXY","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"NO_PROXY","value":".cluster.local,.si-origin-cluster.t-systems.es,.svc,10.49.160.36,172.30.0.1,172.30.248.97,infra.si-origin-cluster.t-systems.es,master.si-origin-cluster.t-systems.es,node1.si-origin-cluster.t-systems.es"},{"name":"http_proxy","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"https_proxy","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"no_proxy","value":".cluster.local,.si-origin-cluster.t-systems.es,.svc,10.49.160.36,172.30.0.1,172.30.248.97,infra.si-origin-cluster.t-systems.es,master.si-origin-cluster.t-systems.es,node1.si-origin-cluster.t-systems.es"}]}},"output":{"to":{"kind":"DockerImage","name":"docker-registry.default.svc:5000/t-devhub-dev/tdev-wsec-service:2.0.3.RELEASE"},"pushSecret":{"name":"builder-dockercfg-t8ktw"}},"resources":{},"postCommit":{},"nodeSelector":{},"triggeredBy":null},"status":{"phase":"New","outputDockerImageReference":"docker-registry.default.svc:5000/t-devhub-dev/tdev-wsec-service:2.0.3.RELEASE","config":{"kind":"BuildConfig","namespace":"t-devhub-dev","name":"tdev-wsec-service-s2i"},"output":{}}}

      ORIGIN_VERSION:  v3.9.0+ba7faec-1
    Mounts:
      /tmp/build from buildworkdir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from builder-token-fz1d5 (ro)
  manage-dockerfile:
    Container ID:  
    Image:         openshift/origin-docker-builder:v3.9.0
    Image ID:      
    Port:          <none>
    Command:
      openshift-manage-dockerfile
    Args:
      --loglevel=0
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:
      BUILD:  {"kind":"Build","apiVersion":"v1","metadata":{"name":"tdev-wsec-service-s2i-1","namespace":"t-devhub-dev","selfLink":"/apis/build.openshift.io/v1/namespaces/t-devhub-dev/builds/tdev-wsec-service-s2i-1","uid":"1291eeec-8e60-11e8-b1f7-0050569fe304","resourceVersion":"50777263","creationTimestamp":"2018-07-23T10:06:31Z","labels":{"app":"tdev-wsec-service","buildconfig":"tdev-wsec-service-s2i","group":"org.springframework.boot","openshift.io/build-config.name":"tdev-wsec-service-s2i","openshift.io/build.start-policy":"Serial","provider":"fabric8","version":"2.0.3.RELEASE"},"annotations":{"openshift.io/build-config.name":"tdev-wsec-service-s2i","openshift.io/build.number":"1"},"ownerReferences":[{"apiVersion":"build.openshift.io/v1","kind":"BuildConfig","name":"tdev-wsec-service-s2i","uid":"11dea374-8e60-11e8-b1f7-0050569fe304","controller":true}]},"spec":{"serviceAccount":"builder","source":{"type":"Binary","binary":{}},"strategy":{"type":"Docker","dockerStrategy":{"env":[{"name":"HTTP_PROXY","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"HTTPS_PROXY","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"NO_PROXY","value":".cluster.local,.si-origin-cluster.t-systems.es,.svc,10.49.160.36,172.30.0.1,172.30.248.97,infra.si-origin-cluster.t-systems.es,master.si-origin-cluster.t-systems.es,node1.si-origin-cluster.t-systems.es"},{"name":"http_proxy","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"https_proxy","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"no_proxy","value":".cluster.local,.si-origin-cluster.t-systems.es,.svc,10.49.160.36,172.30.0.1,172.30.248.97,infra.si-origin-cluster.t-systems.es,master.si-origin-cluster.t-systems.es,node1.si-origin-cluster.t-systems.es"}]}},"output":{"to":{"kind":"DockerImage","name":"docker-registry.default.svc:5000/t-devhub-dev/tdev-wsec-service:2.0.3.RELEASE"},"pushSecret":{"name":"builder-dockercfg-t8ktw"}},"resources":{},"postCommit":{},"nodeSelector":{},"triggeredBy":null},"status":{"phase":"New","outputDockerImageReference":"docker-registry.default.svc:5000/t-devhub-dev/tdev-wsec-service:2.0.3.RELEASE","config":{"kind":"BuildConfig","namespace":"t-devhub-dev","name":"tdev-wsec-service-s2i"},"output":{}}}

      ORIGIN_VERSION:  v3.9.0+ba7faec-1
    Mounts:
      /tmp/build from buildworkdir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from builder-token-fz1d5 (ro)
Containers:
  docker-build:
    Container ID:  
    Image:         openshift/origin-docker-builder:v3.9.0
    Image ID:      
    Port:          <none>
    Command:
      openshift-docker-build
    Args:
      --loglevel=0
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:
      BUILD:  {"kind":"Build","apiVersion":"v1","metadata":{"name":"tdev-wsec-service-s2i-1","namespace":"t-devhub-dev","selfLink":"/apis/build.openshift.io/v1/namespaces/t-devhub-dev/builds/tdev-wsec-service-s2i-1","uid":"1291eeec-8e60-11e8-b1f7-0050569fe304","resourceVersion":"50777263","creationTimestamp":"2018-07-23T10:06:31Z","labels":{"app":"tdev-wsec-service","buildconfig":"tdev-wsec-service-s2i","group":"org.springframework.boot","openshift.io/build-config.name":"tdev-wsec-service-s2i","openshift.io/build.start-policy":"Serial","provider":"fabric8","version":"2.0.3.RELEASE"},"annotations":{"openshift.io/build-config.name":"tdev-wsec-service-s2i","openshift.io/build.number":"1"},"ownerReferences":[{"apiVersion":"build.openshift.io/v1","kind":"BuildConfig","name":"tdev-wsec-service-s2i","uid":"11dea374-8e60-11e8-b1f7-0050569fe304","controller":true}]},"spec":{"serviceAccount":"builder","source":{"type":"Binary","binary":{}},"strategy":{"type":"Docker","dockerStrategy":{"env":[{"name":"HTTP_PROXY","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"HTTPS_PROXY","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"NO_PROXY","value":".cluster.local,.si-origin-cluster.t-systems.es,.svc,10.49.160.36,172.30.0.1,172.30.248.97,infra.si-origin-cluster.t-systems.es,master.si-origin-cluster.t-systems.es,node1.si-origin-cluster.t-systems.es"},{"name":"http_proxy","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"https_proxy","value":"http://user_setools:yGlL2G5V6eGk@10.49.1.1:8080/"},{"name":"no_proxy","value":".cluster.local,.si-origin-cluster.t-systems.es,.svc,10.49.160.36,172.30.0.1,172.30.248.97,infra.si-origin-cluster.t-systems.es,master.si-origin-cluster.t-systems.es,node1.si-origin-cluster.t-systems.es"}]}},"output":{"to":{"kind":"DockerImage","name":"docker-registry.default.svc:5000/t-devhub-dev/tdev-wsec-service:2.0.3.RELEASE"},"pushSecret":{"name":"builder-dockercfg-t8ktw"}},"resources":{},"postCommit":{},"nodeSelector":{},"triggeredBy":null},"status":{"phase":"New","outputDockerImageReference":"docker-registry.default.svc:5000/t-devhub-dev/tdev-wsec-service:2.0.3.RELEASE","config":{"kind":"BuildConfig","namespace":"t-devhub-dev","name":"tdev-wsec-service-s2i"},"output":{}}}

      ORIGIN_VERSION:       v3.9.0+ba7faec-1
      PUSH_DOCKERCFG_PATH:  /var/run/secrets/openshift.io/push
    Mounts:
      /tmp/build from buildworkdir (rw)
      /var/run/crio/crio.sock from crio-socket (rw)
      /var/run/docker.sock from docker-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from builder-token-fz1d5 (ro)
      /var/run/secrets/openshift.io/push from builder-dockercfg-t8ktw-push (ro)
Conditions:
  Type           Status
  Initialized    False 
  Ready          False 
  PodScheduled   True 
Volumes:
  buildworkdir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:  
  docker-socket:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/docker.sock
    HostPathType:  
  crio-socket:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/crio/crio.sock
    HostPathType:  
  builder-dockercfg-t8ktw-push:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  builder-dockercfg-t8ktw
    Optional:    false
  builder-token-fz1d5:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  builder-token-fz1d5
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  zone=tier2
Tolerations:     <none>
Events:
  Type    Reason                 Age   From                                           Message
  ----    ------                 ----  ----                                           -------
  Normal  Scheduled              46m   default-scheduler                              Successfully assigned tdev-wsec-service-s2i-1-build to node4.si-origin-cluster.t-systems.es
  Normal  SuccessfulMountVolume  46m   kubelet, node4.si-origin-cluster.t-systems.es  MountVolume.SetUp succeeded for volume "docker-socket"
  Normal  SuccessfulMountVolume  46m   kubelet, node4.si-origin-cluster.t-systems.es  MountVolume.SetUp succeeded for volume "crio-socket"
  Normal  SuccessfulMountVolume  46m   kubelet, node4.si-origin-cluster.t-systems.es  MountVolume.SetUp succeeded for volume "buildworkdir"
  Normal  SuccessfulMountVolume  46m   kubelet, node4.si-origin-cluster.t-systems.es  MountVolume.SetUp succeeded for volume "builder-token-fz1d5"
  Normal  SuccessfulMountVolume  46m   kubelet, node4.si-origin-cluster.t-systems.es  MountVolume.SetUp succeeded for volume "builder-dockercfg-t8ktw-push"
  Normal  Pulled                 46m   kubelet, node4.si-origin-cluster.t-systems.es  Container image "openshift/origin-docker-builder:v3.9.0" already present on machine
  Normal  Created                46m   kubelet, node4.si-origin-cluster.t-systems.es  Created container
  Normal  Started                46m   kubelet, node4.si-origin-cluster.t-systems.es  Started container
```

Any ideas?

**Comment by jwforres** [comment]
@openshift/sig-developer-experience

**Comment by bparees** [comment]
can you get the logs from the container itself?
`oc logs tdev-wsec-service-s2i-1-build -c git-clone`

also i doubt it matters but you should be able to just run:
`oc start-build --from-archive=docker-build.tar --loglevel=10 forum-s2i --follow`

**Comment by bparees** [comment]
closing based on lack of requested information needed to pursue it.

note that there were some known issues w/ binary builds hanging in 3.10 but that issue did not affect 3.9 (and has been resolved in 3.10 though not made it into an errata release yet).
